fast_llm/models/gpt/conversion.py

```diff
@@ -570,6 +584,110 @@ def _get_mlp_converters(self, fast_llm_prefix: str, hf_prefix: str) -> list[Weig
         ]
 
 
+class MTPLlamaHuggingfaceCheckpointHandler(CommonLlamaHuggingfaceCheckpointHandler):
+    from fast_llm.models.external.mtp_llama import configuration_mtp_llama, modeling_mtp_llama
+
+    format: typing.ClassVar[type[CheckpointFormat]] = MTPLlamaGPTHuggingfaceCheckpointFormat
+    modeling_file = modeling_mtp_llama.__file__
+    configuration_file = configuration_mtp_llama.__file__
+    # modeling_file: typing.ClassVar[str] = "fast_llm/models/external/mtp_llama/modeling_mtp_llama.py"
+    # configuration_file: typing.ClassVar[str] = "fast_llm/models/external/mtp_llama/configuration_mtp_llama.py"
+    configuration_cls = MTPLlamaConfig
+
+    # Use custom config instead of relying on the transformers library
+    @classmethod
```

Comment:
If we're planning on doing this again, might be worth extracting as a generic mixin. The structure looks already good for it.